### PR TITLE
[SYCL] Test to check dependency chain for in-order queue between usm&host tasks

### PR DESCRIPTION
**Pull request by againull**
Test for https://github.com/intel/llvm/pull/6840

### CODE REVIEW

File: SYCL/InorderQueue/in_order_usm_host_dependency.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// Test checks that dependency chain between commands is preserved for in-order
+// queue in the case when usm commands and host tasks are interleaved.
+
+#include <sycl.hpp>
+using namespace sycl;
+
+int main() {
+  queue Q{property::queue::in_order()};
+
+  int *X = malloc_host<int>(100, Q);
```
Comment (KseniyaTikhomirova) [inline SYCL/InorderQueue/in_order_usm_host_dependency.cpp]: minor comment: could we have a constant for size instead raw value in test?

File: SYCL/InorderQueue/in_order_usm_host_dependency.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  Q.memcpy(Y, X, sizeof(int) * 100);
+  Q.copy(Y, Z, 100);
+
+  Q.submit([&](handler &CGH) {
+    auto HostTask = [=] { Z[99] *= 2; };
+    CGH.host_task(HostTask);
+  });
+  Q.wait();
+
+  int Error = (Z[99] != 196) ? 1 : 0;
```
Comment (KseniyaTikhomirova) [inline SYCL/InorderQueue/in_order_usm_host_dependency.cpp]: I guess it should check if host task waits till usm operation completion but could we also check the reverse case? I see that you fill X with values using host task and than overwrite it many times with usm operations but never check and result in final check is also not dependent on it.

File: SYCL/InorderQueue/in_order_usm_host_dependency.cpp (hunk 1 above)
Comment (againull) [reply SYCL/InorderQueue/in_order_usm_host_dependency.cpp]: Fixed.

File: SYCL/InorderQueue/in_order_usm_host_dependency.cpp (hunk 2 above)
Comment (againull) [reply SYCL/InorderQueue/in_order_usm_host_dependency.cpp]: Good call, fixed.

**Review by againull — COMMENTED** [review]
Sorry for abandoning this for a while.

**Comment by againull** [comment]
@KseniyaTikhomirova please review this after addressing your concerns.

**Review by KseniyaTikhomirova — APPROVED** [review]

**Pull request merged**